### PR TITLE
[DoctrineBridge] Don't rely on ClassMetadataInfo->hasField in DoctrineOrmTypeGuesser anymore

### PR DESCRIPTION
**Pull request by fancyweb**
| Q             | A
| ------------- | ---
| Branch?       | 2.7
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | https://github.com/symfony/symfony/issues/26543
| License       | MIT
| Doc PR        | -

Discussion and explanations in the linked issue https://github.com/symfony/symfony/issues/26543

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php [hunk 1]
```diff
@@ -95,7 +95,7 @@ public function guessRequired($class, $property)
         $classMetadata = $classMetadatas[0];
 
         // Check whether the field exists and is nullable or not
-        if ($classMetadata->hasField($property)) {
+        if (isset($classMetadata->fieldMappings[$property])) {
             if (!$classMetadata->isNullable($property) && Type::BOOLEAN !== $classMetadata->getTypeOfField($property)) {
```
Comment (fancyweb) [inline src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php]: `isNullable` calls `getFieldMapping` internally

File: src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php [hunk 2]
```diff
@@ -143,7 +143,7 @@ public function guessMaxLength($class, $property)
     public function guessPattern($class, $property)
     {
         $ret = $this->getMetadata($class);
-        if ($ret && $ret[0]->hasField($property) && !$ret[0]->hasAssociation($property)) {
+        if ($ret && isset($ret[0]->fieldMappings[$property]) && !$ret[0]->hasAssociation($property)) {
```
Comment (fancyweb) [inline src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php]: This one is not needed but I did it for consistency.

**Comment by fancyweb** [comment]
If checking the public property is not clean enough, there are other ways of doing this check.

**Review by Tobion — APPROVED** [review]
LGTM, thanks for the fix

**Review by 20uf — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @fancyweb.

**Comment by pink6440** [comment]
@fancyweb nice and quick work ! thanks !

**Pull request merged**